packages/nimble-components/src/number-field/tests/number-field.foundation.spec.ts

```diff
@@ -0,0 +1,974 @@
+// Based on tests in FAST repo: https://github.com/microsoft/fast/blob/fd9068b94e4aa8d2282f0cce613f58436fae955d/packages/web-components/fast-foundation/src/number-field/number-field.spec.ts
+
+import { DOM } from '@microsoft/fast-element';
+import { NumberField } from '..';
+import { template } from '../template';
+import { fixture } from '../../utilities/tests/fixture';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const FASTNumberField = NumberField.compose({
+    baseName: 'number-field',
+    template
+});
+
+async function setup(props?: Partial<NumberField>): Promise<{
+    element: NumberField,
+    connect: () => Promise<void>,
+    disconnect: () => Promise<void>,
+    parent: HTMLElement
+}> {
+    const { element, connect, disconnect, parent } = await fixture(FASTNumberField());
+
+    if (props) {
+        // eslint-disable-next-line guard-for-in
+        for (const key in props) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-explicit-any
+            (element as any)[key] = (props as any)[key].toString();
+        }
+    }
```

Comment:
Note: This function broke quite a few lint rules we have in place for nimble. I chose to mostly disable those rules and keep the original code rather than completely re-write this function. Let me know if you think it would be better to find a different way to write this code.

Reply:
If you have ideas on another way to write it without disabling the lint rules, it's probably worth doing so (I think we try to fix 'small' lint issues rather than suppress, when we bring FAST tests over).

Reply:
I fixed one of the lint rules, but the collection of issues around assigning an arbitrary property from `props` to `element` seems much more challenging to fix (at least without jumping through a bunch of typing hoops).